### ISSUE TITLE
[feature] can we add keyframes by code?

### ISSUE THREAD

**Issue by moebiussurfing** [roymacdonald/ofxLineaDeTiempo#8]
hey, it's this possible or only from the user GUI?

**Comment by roymacdonald** [comment]
yes. you can add by code.
Thanks for asking cause I just realized how convoluted it is.
A lot of templating things going on. 
I'll add some helper functions which should make it way much easier.

it should be something like 

```
auto region = dynamic_cast<KeyframeRegionController_<float>*>(timeline.getTrack("radius")->getRegion(0));
		if(region)
		{
			region->getKeyframesCollection()[0]->addKeyframe(0.3, 200) ;
		}
```

I'll add right now this simplified feature, as otherwise guessing the templated types can be a real pain, while the classes already "know" such.
I'll let you know

**Comment by smeyfroi** [comment]
Hi Roy, did you get any time to add helpers for setting keyframes in code?

Addon looks really promising so far, just working through getting it up and running here.

**Comment by roymacdonald** [comment]
hi @smeyfroi 
Did you get it to work? with the PR you made it worked for me using OF v0.12.0

I did not, I really cant tell why, probably I just forgot.

It is possible to add keyframes via code right now, it is just very cumbersome.
I will give it a look now and see what comes out.

TL;DR
This addon uses a heavy dose of C++ templates. My original idea was to make it compatible with as much as you wanted but that then became more of a burden than an advantage. I've been thinking for a while to take out some of the templated classes, as at the end the cases in which such becomes useful are not much

**Comment by smeyfroi** [comment]
Thanks @roymacdonald -- yes, I think I'm up and running now: see non-blocker issue just raised.

I do see the generic code in your data layer, but it's quite readable. On the other hand, the casts needed to access any `Data` internals become quite difficult to work with from the outside. So I see your point.

I may have another idea that would work around my use-case. I'll raise it in a different issue in case it turns out to be a simple thing to implement.

**Comment by roymacdonald** [comment]
what is that you are trying to implement?
Maybe I know the workaround.
Right all those casts are quite annoying. I got to work on it, but found some old issues that I decided to tackle (like an ugly bug where the view is all compressed when starting up and it only looks fine when you touch one of the scrollbars.).
As said, removing some of the templating would make it a lot easier, but I have more or less of an idea how to make such helper functions.

**Comment by smeyfroi** [comment]
My usecase is really this: https://github.com/roymacdonald/ofxLineaDeTiempo/issues/16

I think I almost got a hack to work but yes, the templating is confusing me. See that issue... maybe the fix is simple?
